backend/de.metas.business/src/main/java-legacy/org/compiere/model/MInvoiceLine.java

```diff
@@ -1349,6 +1350,27 @@ private boolean updateHeaderTax()
 				throw new AdempiereException("Updating GrandTotal failed; updated records=" + no + "; sql=" + sql);
 			}
 		}
+
+		//
+		// Update Invoice Header: GrandTotal and CashRoundingAmt for 5 cents rounding
```

Comment:
hmmm, now we duplicate the BL in java and SQL.

hmm, i would refactor this method in order to:
* get the total line net amt: `SELECT COALESCE(SUM(LineNetAmt),0) FROM C_InvoiceLine il WHERE i.C_Invoice_ID=il.C_Invoice_ID`
* get the total tax amt: `(SELECT COALESCE(SUM(TaxAmt),0) FROM C_InvoiceTax it WHERE i.C_Invoice_ID=it.C_Invoice_ID AND it.IsActive='Y' AND it.IsTaxIncluded='N')`
* calculate the grand total in java (and apply 5cents rounding if needed)
* run a last SQL which updates C_Invoice's GrandTotal, TotalLines, and CashRoundingAmt